test/samples/plugins/ModularSessionKeyPlugin.t.sol

```diff
@@ -0,0 +1,381 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {Test} from "forge-std/Test.sol";
+
+import {EntryPoint} from "@eth-infinitism/account-abstraction/core/EntryPoint.sol";
+import {UserOperation} from "@eth-infinitism/account-abstraction/interfaces/UserOperation.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+import {SingleOwnerPlugin} from "../../../src/plugins/owner/SingleOwnerPlugin.sol";
+import {ISingleOwnerPlugin} from "../../../src/plugins/owner/ISingleOwnerPlugin.sol";
+import {ModularSessionKeyPlugin} from "../../../src/samples/plugins/ModularSessionKeyPlugin.sol";
+import {ISessionKeyPlugin} from "../../../src/samples/plugins/interfaces/ISessionKeyPlugin.sol";
+import {TokenSessionKeyPlugin} from "../../../src/samples/plugins/TokenSessionKeyPlugin.sol";
+import {ITokenSessionKeyPlugin} from "../../../src/samples/plugins/interfaces/ITokenSessionKeyPlugin.sol";
+
+import {UpgradeableModularAccount} from "../../../src/account/UpgradeableModularAccount.sol";
+import {MSCAFactoryFixture} from "../../mocks/MSCAFactoryFixture.sol";
+import {FunctionReference, FunctionReferenceLib} from "../../../src/helpers/FunctionReferenceLib.sol";
+import {IPluginManager} from "../../../src/interfaces/IPluginManager.sol";
+import {MockERC20} from "../../mocks/MockERC20.sol";
+
+contract ModularSessionKeyPluginTest is Test {
+    using ECDSA for bytes32;
+    using FunctionReferenceLib for address;
```

Comment:
Nit: I feel like using `pack` this way is a bit odd. I'd prefer `FunctionReferenceLib.pack(a, b)` instead of `a.pack(b)`.